### PR TITLE
mod: don't rate for non advertisers

### PR DESCRIPTION
**Pull request by ruibinx**

### CODE REVIEW

File: pallets/ad/src/tests.rs [hunk 1]
```diff
@@ -1175,3 +1178,54 @@ fn should_not_reward_if_score_is_negative() {
         assert_eq!(Assets::balance(nft_meta.token_asset_id, &CHARLIE), balance);
     });
 }
+
+#[test]
+pub fn non_advertisers_should_not_affect_rating() {
```
Comment (coderfengyun) [inline pallets/ad/src/tests.rs]: 加一个测试用例non_advertiser_should_affect_rating_when_diff_score_is_negative

Comment (coderfengyun) [inline pallets/ad/src/tests.rs]:
```suggestion
pub fn non_advertisers_should_not_affect_rating_when_score_diff_is_positive() {
```

File: pallets/advertiser/src/lib.rs [hunk 2]
```diff
@@ -159,24 +159,30 @@ pub mod pallet {
             }
         }
     }
+
+    impl<T: Config> Pallet<T> {
+        pub fn is_account_advertiser(who: &AccountOf<T>) -> bool {
```
Comment (coderfengyun) [inline pallets/advertiser/src/lib.rs]:
```suggestion
        pub fn is_advertiser(who: &AccountOf<T>) -> bool {
```

File: pallets/ad/src/tests.rs (hunk 1 above)
Comment (ruibinx) [reply pallets/ad/src/tests.rs]: 跟claim_advertiser_without_signature的用例是重的。

Comment (coderfengyun) [reply pallets/ad/src/tests.rs]: 走带signature的接口也可能会打出负分

File: pallets/ad/src/tests.rs [hunk 3]
```diff
@@ -1175,3 +1178,105 @@ fn should_not_reward_if_score_is_negative() {
...
+
+        assert_eq!(
+            Tag::get_score(&DID_CHARLIE, vec![0u8, 1u8, 2u8, 3u8, 4u8, 5u8]),
+            5
+        );
+    });
+}
+
+#[test]
+pub fn non_advertisers_should_not_affect_ratin_when_score_diff_is_negative() {
```
Comment (coderfengyun) [inline pallets/ad/src/tests.rs:1234]:
```suggestion
pub fn non_advertisers_should_affect_ratin_when_score_diff_is_negative() {
```

**Review by coderfengyun — APPROVED** [review]

**Review by coderfengyun — APPROVED** [review]

**Pull request merged**